# HG changeset patch
# User Andreas Weber <octave@josoansi.de>
# Date 1546553358 -3600
#      Thu Jan 03 23:09:18 2019 +0100
# Branch stable
# Node ID ab6b8d8e0307fc861b7683b28a81f7a015b7c59a
# Parent  d644dec272ea2e2e9e9b49fb4dd0912f3b67c54e
oct-stream.cc: Fix static analyzer detected V519 issue (bug #55347).

diff --git a/libinterp/corefcn/oct-stream.cc b/libinterp/corefcn/oct-stream.cc
--- a/libinterp/corefcn/oct-stream.cc
+++ b/libinterp/corefcn/oct-stream.cc
@@ -4602,17 +4602,16 @@ namespace octave
         return Matrix (nr, nc, 0.0);
       }
 
     std::istream *isp = input_stream ();
 
     bool all_char_conv = fmt_list.all_character_conversions ();
 
     Matrix mval;
-    double *data = nullptr;
     octave_idx_type max_size = 0;
     octave_idx_type max_conv = 0;
 
     octave_idx_type final_nr = 0;
     octave_idx_type final_nc = 0;
 
     if (all_char_conv)
       {
@@ -4660,17 +4659,17 @@ namespace octave
       }
     else
       {
         // Maybe resize later.
         mval.resize (32, 1, 0.0);
         max_size = 32;
       }
 
-    data = mval.fortran_vec ();
+    double *data = mval.fortran_vec ();
 
     if (isp)
       {
         std::istream& is = *isp;
 
         const scanf_format_elt *elt = fmt_list.first ();
 
         std::ios::fmtflags flags = is.flags ();
